library/pkcs5.c

```diff
@@ -230,9 +231,13 @@ int mbedtls_pkcs5_pbkdf2_hmac( mbedtls_md_context_t *ctx, const unsigned char *p
 
     memset( counter, 0, 4 );
     counter[3] = 1;
-
+/*
+ * relevant only for 64 bit platforms. On 32 bit platforms, unsigned int cannot exceed 32 bits
+ */
+#if defined(MBEDTLS_HAVE_INT64)
     if( iteration_count > 0xFFFFFFFF )
```

Comment:
I think this check is redundant as `iteration_count` is of type `unsigned int`. In my 64-bit machine this is still a 32-bit integer. Howeber, I am aware that this is entirely compiler dependent as the C standard only enforces minimum values that need to be held by each type.

What is your opinion on this?